runtime/compiler/trj9/ilgen/Walker.cpp

```diff
@@ -7037,7 +7053,7 @@ TR_J9ByteCodeIlGenerator::genMultiANewArray(int32_t dims)
 //----------------------------------------------
 
 int32_t
-TR_J9ByteCodeIlGenerator::genReturn(TR::ILOpCodes nodeop, bool monitorExit)
+TR_J9ByteCodeIlGenerator::genReturn(TR::ILOpCodes nodeop, TR::ILOpCodes truncationConversion, TR::ILOpCodes extensionConversion, bool isZConversion, bool monitorExit)
```

Comment:
Rather than polluting the genReturn call with flags why can't we just use the bytecode with a switch statement in the genReturn method to set these conversion opcodes? It will make the method much more self contained... We are trying to avoid Boolean flags on APIs as much as possible.

Reply:
I completely agree. The updated API indeed looks much better. Fixed in 8d8a7f4